packages/concerto-tools/lib/codegen/fromcto/jsonschema/jsonschemavisitor.js

```diff
@@ -453,6 +473,41 @@ class JSONSchemaVisitor {
         return jsonSchema;
     }
 
+    /**
+     * Visitor a scalar declaration
+     * @param {ScalarDeclaration} scalarDeclaration - the scalar declaration being visited
+     * @param {Object} parameters - the parameter
+     * @return {Object} the result of visiting or null
+     * @private
+     */
+    visitScalarDeclaration(scalarDeclaration, parameters) {
```

Comment:
I'm assuming we will have to add similar logic to all the code generators? I think it might be very useful to add a method to the scalar class to convert it to its equivalent Property, so people can update the generators using something like:

```
else if (thing.isScalarDeclaration?.()) {
             return this.visitProperty(thing.toProperty(), parameters);
```

Effectively calling the code they already have that can handle a property. Make sense?

Reply:
Some generators may be able to support scalars natively (GraphQL fe) but most will not.